test/distributed/partition_helper_kernels.cpp

```diff
@@ -0,0 +1,279 @@
+                                       gko::size_type n)
+{
+    std::default_random_engine engine;
+    auto values = create_iota(min, max);
+    std::shuffle(values.begin(), values.end(), engine);
+    values.erase(values.begin() + clamp(n, gko::size_type(0), values.size()),
+                 values.end());
+    return values;
+}
+
+
+template <typename IndexType>
+std::vector<IndexType> remove_indices(const std::vector<IndexType>& source,
+                                      std::vector<std::size_t> idxs)
+{
+    std::sort(idxs.begin(), idxs.end(), std::greater<>{});
+    auto result = source;
+    for (auto idx : idxs) {
+        result.erase(result.begin() + 2 * idx, result.begin() + 2 * idx + 1);
+    }
+    return result;
+}
+
+
+template <typename IndexType>
+gko::array<IndexType> make_array(std::shared_ptr<const gko::Executor> exec,
+                                 const std::vector<IndexType>& v)
+{
+    return gko::array<IndexType>(exec, v.begin(), v.end());
+}
+
+
+template <typename IndexType>
+std::pair<std::vector<IndexType>, std::vector<comm_index_type>>
+shuffle_range_and_pid(const std::vector<IndexType>& ranges,
+                      const std::vector<comm_index_type>& pid)
+{
+    struct range {
+        IndexType vals[2];
+    };
+
+    std::default_random_engine engine;
+
+    auto result = std::make_pair(ranges, pid);
+
+    auto num_ranges = result.second.size();
+    auto zip_it = gko::detail::make_zip_iterator(
+        reinterpret_cast<range*>(result.first.data()), result.second.begin());
+    std::shuffle(zip_it, zip_it + num_ranges, engine);
+
+    return result;
+}
+
+template <typename IndexType>
+class PartitionHelpers : public CommonTestFixture {
+protected:
+    using index_type = IndexType;
+};
+
+TYPED_TEST_SUITE(PartitionHelpers, gko::test::IndexTypes);
+
+
+TYPED_TEST(PartitionHelpers, CanCheckConsecutiveRanges)
+{
+    using index_type = typename TestFixture::index_type;
+    auto offsets = make_array(this->exec, create_ranges<index_type>(100));
+    bool result = false;
+
+    gko::kernels::EXEC_NAMESPACE::partition_helpers::check_consecutive_ranges(
+        this->exec, offsets, &result);
+
+    ASSERT_TRUE(result);
+}
+
+
+TYPED_TEST(PartitionHelpers, CanCheckNonConsecutiveRanges)
+{
+    using index_type = typename TestFixture::index_type;
+    auto full_range_ends = create_ranges<index_type>(100);
+    auto removal_idxs = sample_unique(0, full_range_ends.size() / 2, 4);
+    auto start_ends =
+        make_array(this->exec, remove_indices(full_range_ends, removal_idxs));
+    bool result = true;
+
+    gko::kernels::EXEC_NAMESPACE::partition_helpers::check_consecutive_ranges(
+        this->exec, start_ends, &result);
+
+    ASSERT_FALSE(result);
+}
+
+
+ TYPED_TEST(PartitionHelpers, CanCheckConsecutiveRangesWithSingleRange)
+{
+    using index_type = typename TestFixture::index_type;
+    auto start_ends =
+        make_array(this->ref, create_ranges<index_type>(1));
```

Comment:
This should not work properly (since `begin+1 < end - 1`).

Reply:
What part should not work? This creates a temporary vector with two elements, so `begin + 1 == end - 1`.

Reply:
My bad, I thought it would create a 1-element array (number of arguments) instead of 1+1.